jcl/src/java.base/share/classes/openj9/management/internal/ThreadInfoBase.java

```diff
@@ -0,0 +1,398 @@
+		nativeTId = nativeId;
+		threadName = thread.getName();
+		/*[IF Sidecar19-SE]*/
+		daemon = thread.isDaemon();
+		priority = thread.getPriority();
+		/*[ENDIF]*/
+
+		// Use the supplied state int to index into the array returned
+		// by values() which should be all thread states in the order
+		// they have been declared. Doing this rather than just issuing a
+		// call to thread.getState() because the Thread state may have
+		// changed after the VM issued the call to this method.
+		threadState = Thread.State.values()[state];
+		suspended = isSuspended;
+		inNative = isInNative;
+		blockedCount = blockedCountVal;
+		blockedTime = blockedTimeVal;
+		waitedCount = waitedCountVal;
+		waitedTime = waitedTimeVal;
+
+		if (blockingObjectOwner != null) {
+			lockOwnerId = blockingObjectOwner.getId();
+			lockOwnerName = blockingObjectOwner.getName();
+		} else {
+			lockOwnerId = -1;
+			lockOwnerName = null;
+		}
+
+		stackTrace = stackTraceVal;
+		String lockNameTemp = null;
+		if (blockingObject != null) {
+			lockNameTemp = blockingObject.getClass().getName() + '@'
+					+ Integer.toHexString(System.identityHashCode(blockingObject));
+		}
+
+		lockedMonitors = lockedMonitorsVal;
+		lockedSynchronizers = lockedSynchronizersVal;
+
+		if (blockingObject != null) {
+			blockingLockInfo = new LockInfoBase(blockingObject.getClass().getName(),
+					System.identityHashCode(blockingObject));
+		} else {
+			blockingLockInfo = null;
+		}
+		if (null != blockingLockInfo) {
+			lockNameTemp = blockingLockInfo.toString();
+		}
+		lockName = lockNameTemp;
+	}
+
+	/**
+	 * @return a text description of this thread info.
+	 */
+	@Override
+	public String toString() {
+		/* Since ThreadInfos are immutable the string value need be calculated only once. */
+		String ls = System.lineSeparator();
+		if (TOSTRING_VALUE == null) {
+			StringBuilder result = new StringBuilder();
+/*[IF Java11]*/
+			result.append(String.format("\"%s\" prio=%d Id=%d %s", //$NON-NLS-1$
+					threadName, Integer.valueOf(priority), Long.valueOf(threadId), threadState));
+/*[ELSE]*/
+			result.append(String.format("\"%s\" Id=%d %s", //$NON-NLS-1$
+					threadName, Long.valueOf(threadId), threadState));
+/*[ENDIF]*/
+			if (State.BLOCKED == threadState) {
+				result.append(String.format(" on %s owned by \"%s\" Id=%d", //$NON-NLS-1$
+						lockName, lockOwnerName, Long.valueOf(lockOwnerId)));
+			}
+			result.append(ls);
+			if ((stackTrace != null) && (stackTrace.length > 0)) {
+				MonitorInfoBase[] lockArray = new MonitorInfoBase[stackTrace.length];
+				for (MonitorInfoBase mi : lockedMonitors) {
+					int lockDepth = mi.stackDepth;
+					lockArray[lockDepth] = mi;
+				}
+				int stackDepth = 0;
+				for (StackTraceElement element : stackTrace) {
+					result.append("\tat "); //$NON-NLS-1$
+					Util.printStackTraceElement(element, null, result, true);
+					result.append(ls);
+					if (null != lockArray[stackDepth]) {
+						MonitorInfoBase mi = lockArray[stackDepth];
```

Comment:
Pull this declaration outside the `if` statement so the value can be reused in the `if` condition.